syntax = "proto3";
package reservation;

option csharp_namespace = "Reservation.Protos";

service ReservationGrpc {
    rpc CreateReservation (CreateReqeust) returns (ReservationResponse);
    rpc CancelReservation (ChangeReqeust) returns (ReservationResponse);
    rpc CompleteReservation (ChangeReqeust) returns (ReservationResponse);
    rpc RetriveUsersReservations (RetriveRequest) returns (ReservationResponse);
}

message CreateReqeust {
    string item_id = 1;
    string user_id = 2;
    Format item_format = 3;
}

message ChangeReqeust {
    string id = 1;
}

message RetriveRequest {
    string user_id = 1;
}

message ReservationResponse {
    string id = 1;
    string item_id = 2;
    string user_id = 3;
    Format item_format = 4;
    string expiry_time_unix = 5;
}

message RetriveResponse {
    repeated ReservationResponse reservations = 1;
}

enum Format {
    NULL = 0;
    BOOK = 1;
    VINYL = 2;
}